state-chain/pallets/cf-elections/src/electoral_systems/egress_success.rs

```diff
@@ -0,0 +1,115 @@
+use crate::{
+	electoral_system::{
+		AuthorityVoteOf, ElectionIdentifierOf, ElectionReadAccess, ElectionWriteAccess,
+		ElectoralSystem, ElectoralWriteAccess, VotePropertiesOf,
+	},
+	vote_storage::{self, VoteStorage},
+	CorruptStorageError,
+};
+use cf_primitives::AuthorityCount;
+use cf_utilities::{all_same, success_threshold_from_share_count};
+use frame_support::{
+	pallet_prelude::{MaybeSerializeDeserialize, Member},
+	Parameter,
+};
+use sp_std::vec::Vec;
+
+/// This electoral system detects if something occurred or not. Voters simply vote if something
+/// happened, and if they haven't seen it happen, they don't vote.
+pub struct EgressSuccess<Identifier, Value, Settings, Hook> {
+	_phantom: core::marker::PhantomData<(Identifier, Value, Settings, Hook)>,
+}
+
+pub trait OnEgressSuccess<Identifier, Value> {
+	fn on_egress_success(id: Identifier, value: Value);
+}
+
+impl<
+		Identifier: Member + Parameter + Ord,
+		Value: Member + Parameter + Eq,
+		Settings: Member + Parameter + MaybeSerializeDeserialize + Eq,
+		Hook: OnEgressSuccess<Identifier, Value> + 'static,
+	> EgressSuccess<Identifier, Value, Settings, Hook>
+{
+	pub fn watch_for_egress<ElectoralAccess: ElectoralWriteAccess<ElectoralSystem = Self>>(
```

Comment:
This is also very similar to `watch_for_change`.

In general a lot of these impls are very similar...